File: KamiFileSystem/src/com/kagami/filesystem/ser/SettingServlet.java
package com.kagami.filesystem.ser;

import org.json.JSONObject;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * Created by sinceredeveloper on 16/12/20.
 */
@WebServlet(name = "SettingServlet",urlPatterns = "setting")
public class SettingServlet extends HttpServlet {
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

    }

    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        JSONObject ret=new JSONObject();
        ret.put("path","/////");
        response.addHeader("Access-Control-Allow-Origin","*");
        response.getWriter().write(ret.toString());
    }
}

File: KamiFileSystem/src/com/kagami/filesystem/ser/SettingServlet.kt
package com.kagami.filesystem.ser

import com.kagami.filesystem.utils.DBManager
import com.kagami.filesystem.utils.FileManager
import java.io.File
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * Created by sinceredeveloper on 16/12/30.
 */
@WebServlet(name = "SettingServlet", urlPatterns = arrayOf("setting"))
class SettingServlet: HttpServlet(){
    override fun doGet(req: HttpServletRequest?, resp: HttpServletResponse) {
        resp.addHeader("Access-Control-Allow-Origin", "*")
        resp.getWriter().write(DBManager.setting.toString())
    }

    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse?) {
        val roodir=req.getParameter("rootdir")
        val dlurl=req.getParameter("dlurl")
        DBManager.putSetting("rootdir",roodir)
                .putSetting("dlurl",dlurl)
                .saveSetting()
        FileManager.setupWithDic(File(roodir))
    }
}
